src/basis/type/enumeration.js

```diff
@@ -0,0 +1,44 @@
+function enumeration(values) {
+  if (!Array.isArray(values))
+  {
+    /** @cut */ basis.dev.warn('basis.type.enum constructor expected array but got ' + values + '. Wrapping into array');
+    values = [values];
+  }
+
+  if (!values.length)
+  {
+    /** @cut */ basis.dev.warn('basis.type.enum constructor expected non-empty array but got empty. Falling back to [null]');
+    values = [null];
```

Comment:
I guess `transform` will always returns `undefined` that's correct in case of empty array.
Another solution is throw exception on non-array or empty array values.